src/cplus_plugin/gui/constant_raster_manager_dialog.py

```diff
@@ -0,0 +1,1135 @@
+                # Item is checked - ensure it has a component
+                if component is None:
+                    # Create a new component
+                    component = current_config_widget.create_raster_component(
+                        item.model_component
+                    )
+                    raster_collection.components.append(component)
+                    items_created.append(item.text())
+                else:
+                    items_updated.append(item.text())
+
+                # Link the model component if not already set
+                if component.component is None:
+                    component.component = item.model_component
+                    component.alias_name = item.text()
+
+                # Apply the current widget value to newly created components
+                # Existing components keep their saved values
+                if items_created and item.text() in items_created:
+                    # This is a new component, apply current widget value
+                    if current_value is not None and component.value_info:
+                        component.value_info.absolute = current_value
+
+                # Mark as enabled
+                component.skip_raster = False
+            else:
+                # Item is unchecked - mark as disabled if it exists
+                if component is not None:
+                    component.skip_raster = True
+
+        # Info message about what was done
+        if items_created or items_updated:
+            msg_parts = []
+            if items_created:
+                msg_parts.append(f"Created {len(items_created)} component(s)")
+            if items_updated:
+                msg_parts.append(f"Updated {len(items_updated)} component(s)")
+            msg = " and ".join(msg_parts)
+
+            if current_value is not None:
+                self.message_bar.pushInfo(
+                    "Components Configured", f"{msg} with value: {current_value}"
+                )
+            else:
+                self.message_bar.pushWarning(
+                    "Using Default Values",
+                    f"{msg} with default value (0.0). Enter a value in the widget first.",
+                )
+
+        # NOTE: We don't call normalize() here anymore because:
+        # 1. The user specifies min/max values explicitly in the UI
+        # 2. Auto-normalizing would override user values
+        # 3. With only one component, min==max causes incorrect 0.5 fallback
+
+    def _get_pixel_size_from_snap_layer(self) -> float:
+        """Get pixel size from the snap layer for consistency with analysis.
+
+        The snap layer is used as a reference to align all rasters during
+        scenario analysis. Using its pixel size ensures constant rasters
+        are consistent with other processing outputs.
+
+        :returns: Pixel size in map units, defaults to 30.0 if snap layer unavailable
+        """
+        DEFAULT_PIXEL_SIZE = 30.0
```

Comment:
@osundwajeff we can already get the pixel size in two ways:
1. If its an NCS pathway, we can use the ID to get the NCS pathway from settings and load the raster to get its x/y size. This will be uniform since the validation checks if the NCS pathways have the same resolution.
2. If its an activity, we can get the first NCS pathway in the activity then use the procedure in (1) above to get the resolution of the NCS pathway.

Having a default here may not be ideal since it makes it difficult to heuristically resolve whether the units are for geographic or projected coordinates.